django_project/stakeholder/views.py

```diff
@@ -68,3 +86,395 @@ def get_context_data(self, **kwargs):
 
         return context
+
+
+def create_return_results(reminders):
+    """this methods constructs the object to return
+    to the front end for search tables and crud operations
+    on both notifications and reminders"""
+    search_results = []
+    for reminder in reminders:
+        search_results.append(
+            {
+                'id': reminder.id,
+                'title': reminder.title,
+                'reminder': reminder.reminder,
+                'user': str(reminder.user),
+                'organisation': str(reminder.organisation),
+                'status': reminder.status,
+                'date': datetime.strptime(
+                    str(
+                        reminder.date + timedelta(hours=2)),
+                    "%Y-%m-%d %H:%M:%S%z"
+                ).strftime("%Y-%m-%d %I:%M %p"),
+                'type': reminder.type,
+                'email_sent': reminder.email_sent
+            }
+        )
+    return search_results
+
+
+def search_reminders_or_notifications(request):
+    search_query = request.POST.get('query')
+    filter = request.POST.get('filter')
+    notifications_page = request.POST.get('notifications_page')
+    if filter is not None and filter != '':
+        if filter == 'title':
+            reminders = Reminders.objects.filter(
+                Q(user=request.user),
+                Q(organisation=request.session[CURRENT_ORGANISATION_ID_KEY]),
+                Q(title__icontains=search_query)
+            )
+        else:
+            reminders = Reminders.objects.filter(
+                Q(user=request.user),
+                Q(organisation=request.session[CURRENT_ORGANISATION_ID_KEY]),
+                Q(reminder__icontains=search_query)
+            )
+    else:
+        reminders = Reminders.objects.filter(
+            Q(user=request.user),
+            Q(organisation=request.session[CURRENT_ORGANISATION_ID_KEY]),
+            Q(title__icontains=search_query) | Q(
+                reminder__icontains=search_query)
+        )
+    if notifications_page is not None:
+        notifications = []
+        for reminder in reminders:
+            if reminder.status == Reminders.PASSED and reminder.email_sent:
+                notifications.append(reminder)
+        return notifications
+    else:
+        return reminders
+
+
+def delete_reminder_and_notification(request):
+    data = json.loads(request.POST.get('ids'))
+    try:
+        for element in data:
+            if isinstance(element, str) and element.isdigit():
+                Reminders.objects.filter(
+                    user=request.user,
+                    organisation=request.session[CURRENT_ORGANISATION_ID_KEY],
+                    id=int(element)
+                ).delete()
+
+        reminders = Reminders.objects.filter(
+            user=request.user,
+            organisation=request.session[CURRENT_ORGANISATION_ID_KEY]
+        )
+        return reminders
+    except Exception as e:
+        print(str(e))
```

Comment:
This is not a good practice, as it means the program may continue executing even after an error has occurred.

Reply:
We should handle the exception, perhaps raise show error to the user.